homeassistant/components/metoffice/weather.py

```diff
@@ -65,138 +83,222 @@ async def async_setup_entry(
+        ATTR_FORECAST_PRECIPITATION_PROBABILITY
+    )
+    forecast[ATTR_FORECAST_UV_INDEX] = get_mapped_attribute(ATTR_FORECAST_UV_INDEX)
+    forecast[ATTR_FORECAST_WIND_BEARING] = get_mapped_attribute(
+        ATTR_FORECAST_WIND_BEARING
+    )
+    forecast[ATTR_FORECAST_NATIVE_WIND_SPEED] = get_mapped_attribute(
+        ATTR_FORECAST_NATIVE_WIND_SPEED
+    )
+    forecast[ATTR_FORECAST_NATIVE_WIND_GUST_SPEED] = get_mapped_attribute(
+        ATTR_FORECAST_NATIVE_WIND_GUST_SPEED
+    )
 
 
 class MetOfficeWeather(
     CoordinatorWeatherEntity[
-        TimestampDataUpdateCoordinator[MetOfficeData],
-        TimestampDataUpdateCoordinator[MetOfficeData],
+        TimestampDataUpdateCoordinator[ForecastData],
+        TimestampDataUpdateCoordinator[ForecastData],
+        TimestampDataUpdateCoordinator[ForecastData],
     ]
 ):
     """Implementation of a Met Office weather condition."""
 
     _attr_attribution = ATTRIBUTION
     _attr_has_entity_name = True
+    _attr_name = None
 
     _attr_native_temperature_unit = UnitOfTemperature.CELSIUS
-    _attr_native_pressure_unit = UnitOfPressure.HPA
-    _attr_native_wind_speed_unit = UnitOfSpeed.MILES_PER_HOUR
+    _attr_native_pressure_unit = UnitOfPressure.PA
+    _attr_native_precipitation_unit = UnitOfLength.MILLIMETERS
+    _attr_native_visibility_unit = UnitOfLength.METERS
+    _attr_native_wind_speed_unit = UnitOfSpeed.METERS_PER_SECOND
     _attr_supported_features = (
-        WeatherEntityFeature.FORECAST_HOURLY | WeatherEntityFeature.FORECAST_DAILY
+        WeatherEntityFeature.FORECAST_HOURLY
+        | WeatherEntityFeature.FORECAST_TWICE_DAILY
+        | WeatherEntityFeature.FORECAST_DAILY
     )
 
     def __init__(
         self,
-        coordinator_daily: TimestampDataUpdateCoordinator[MetOfficeData],
-        coordinator_hourly: TimestampDataUpdateCoordinator[MetOfficeData],
+        coordinator_daily: TimestampDataUpdateCoordinator[ForecastData],
+        coordinator_twice_daily: TimestampDataUpdateCoordinator[ForecastData],
+        coordinator_hourly: TimestampDataUpdateCoordinator[ForecastData],
         hass_data: dict[str, Any],
     ) -> None:
         """Initialise the platform with a data instance."""
-        observation_coordinator = coordinator_daily
+        observation_coordinator = coordinator_hourly
         super().__init__(
             observation_coordinator,
             daily_coordinator=coordinator_daily,
+            twice_daily_coordinator=coordinator_twice_daily,
             hourly_coordinator=coordinator_hourly,
         )
 
         self._attr_device_info = get_device_info(
             coordinates=hass_data[METOFFICE_COORDINATES], name=hass_data[METOFFICE_NAME]
         )
-        self._attr_name = "Daily"
-        self._attr_unique_id = _calculate_unique_id(
-            hass_data[METOFFICE_COORDINATES], False
-        )
+        self._attr_unique_id = hass_data[METOFFICE_COORDINATES]
```

Comment:
This would create conflicts as all entities would get the same unique id?

Reply:
This is a `weather` entity, there is only one being provided.